client/src/nv_ingest_client/client/job_manager.py

```diff
@@ -0,0 +1,348 @@
+
+        Parameters
+        ----------
+        kwargs : dict
+            Additional parameters for the `submit_job_async` method.
+
+        Returns
+        -------
+        Future
+            A future that completes when all submitted jobs have reached a terminal state.
+        """
+        self._job_ids = self._client.add_job(self._job_specs)
+        future_to_job_id = self._client.submit_job_async(self._job_ids, self._job_queue_id, **kwargs)
+        self._job_states = {job_id: self._client._get_and_check_job_state(job_id) for job_id in self._job_ids}
+
+        combined_future = Future()
+        submitted_futures = set(future_to_job_id.keys())
+        completed_futures = set()
+        future_results = []
+
+        def _done_callback(future):
+            job_id = future_to_job_id[future]
+            job_state = self._job_states[job_id]
+            try:
+                result = self._client.fetch_job_result(job_id)
+                if job_state.state != JobStateEnum.COMPLETED:
+                    job_state.state = JobStateEnum.COMPLETED
+            except Exception:
+                result = None
+                if job_state.state != JobStateEnum.FAILED:
+                    job_state.state = JobStateEnum.FAILED
+            completed_futures.add(future)
+            future_results.append(result)
+            if completed_futures == submitted_futures:
+                combined_future.set_result(future_results)
+
+        for future in future_to_job_id:
+            future.add_done_callback(_done_callback)
+
+        return combined_future
+
+    def dedup(self, **kwargs: Any) -> "NvIngestJobManager":
+        """
+        Adds a DedupTask to the batch job specification.
+
+        Parameters
+        ----------
+        kwargs : dict
+            Parameters specific to the DedupTask.
+
+        Returns
+        -------
+        NvIngestJobManager
+            Returns self for chaining.
+        """
+        dedup_task = DedupTask(**kwargs)
+        self._job_specs.add_task(dedup_task)
+
+        return self
+
+    def embed(self, **kwargs: Any) -> "NvIngestJobManager":
+        """
+        Adds an EmbedTask to the batch job specification.
+
+        Parameters
+        ----------
+        kwargs : dict
+            Parameters specific to the EmbedTask.
+
+        Returns
+        -------
+        NvIngestJobManager
+            Returns self for chaining.
+        """
+        embed_task = EmbedTask(**kwargs)
+        self._job_specs.add_task(embed_task)
+
+        return self
+
+    def extract(self, **kwargs: Any) -> "NvIngestJobManager":
+        """
+        Adds an ExtractTask for each document type to the batch job specification.
+
+        Parameters
+        ----------
+        kwargs : dict
+            Parameters specific to the ExtractTask.
+
+        Returns
+        -------
+        NvIngestJobManager
+            Returns self for chaining.
+        """
+        for document_type in self._job_specs.file_types:
```

Comment:
Please update this to add `table_data_extract` and `chart_data_extract` tasks if extract is specified with `extract_tables` and/or `extract_charts`.